corehq/apps/export/views.py

```diff
@@ -296,113 +407,804 @@ def create_basic_form_checkpoint(index):
+        }
+        """
+        try:
+            export_filter, export_specs = self._process_filters_and_specs(in_data)
+            if len(export_specs) > 1:
+                download = self._get_bulk_download_task(export_specs, export_filter)
+            else:
+                max_column_size = int(in_data.get('max_column_size', 2000))
+                download = self._get_download_task(
+                    export_specs, export_filter, max_column_size
+                )
+        except ExportAsyncException as e:
+            return format_angular_error(e.message)
+        except Exception as e:
+            return format_angular_error(
+                e.message,
+                log_error=True,
+                exception=e,
+                request=self.request,
+            )
+        return format_angular_success({
+            'download_id': download.download_id,
+        })
+
+
+class DownloadFormExportView(BaseDownloadExportView):
+    """View to download a SINGLE Form Export with filters.
+    """
+    urlname = 'export_download_forms'
+    show_date_range = True
+    page_title = ugettext_noop("Download Form Export")
+    check_for_multimedia = True
+
+    @staticmethod
+    def get_export_schema(export_id):
+        return FormExportSchema.get(export_id)
+
+    @property
+    def export_list_url(self):
+        return reverse(FormExportListView.urlname, args=(self.domain,))
 
     @property
     @memoized
-    def create_export_form(self):
-        if self.request.method == 'POST':
-            return CreateFormExportForm(self.domain, self.request.POST)
-        return CreateFormExportForm(self.domain)
+    def download_export_form(self):
+        return FilterFormExportDownloadForm(
+            self.domain_object,
+            self.timezone,
+            initial={
+                'type_or_group': 'type',
+            },
+        )
 
     @property
-    def app_to_module_options(self):
+    def parent_pages(self):
+        if not self.has_edit_permissions:
+            return [{
+                'title': DeIdFormExportListView.page_title,
+                'url': reverse(DeIdFormExportListView.urlname, args=(self.domain,)),
+            }]
+        return [{
+            'title': FormExportListView.page_title,
+            'url': reverse(FormExportListView.urlname, args=(self.domain,)),
+        }]
+
+    def get_filters(self, filter_form_data):
+        filter_form = FilterFormExportDownloadForm(
+            self.domain_object, self.timezone, filter_form_data)
+        if not filter_form.is_valid():
+            raise ExportFormValidationException()
+        form_filter = filter_form.get_form_filter()
+        export_filter = SerializableFunction(default_form_filter,
+                                             filter=form_filter)
+        return export_filter
+
+    def get_export_object(self, export_id):
+        return FormExportSchema.get(export_id)
+
+    def _get_attachment_size_by_app_id(self):
```

Comment:
FYI, once https://github.com/dimagi/commcare-hq/pull/8921 gets merged you can use the `get_attachment_size_by_domain_app_id_xmlns` function to only get the data for the `app_id`, `xmlns` combo you need.